latex/ledger-spec.tex

```diff
@@ -1431,27 +1393,16 @@ \subsection{Witnesses}
     \label{eq:utxo-witness-inductive}
     \inference[UTxO-wit]
     {
+      (utxo, \wcard, \wcard) = \var{utxoSt} \\~\\
+      \forall \var{vk} \mapsto \sigma \in \txwits{tx},
+        \mathcal{V}_{\var{vk}}{\serialised{\txbody{tx}}}_{\sigma} \\
+      \fun{witsNeeded}~{utxo}~{tx} = \{ \hashKey \var{vk} \mid \var{vk}\in\dom{\txwits{tx}} \}\\
```

Comment:
If I understand this correctly here we're saying that the set of keys associated to a witness must coincide with the keys that need witnesses, right? This is what would ensure that we don't have any junk, and that everything that needs signing is signed.

Reply:
Why not replacing `\{ \hashKey \var{vk} \mid \var{vk}\in\dom{\txwits{tx}} \}` with `\dom{(\txwits{tx})}` (I think these should be equivalent).

Reply:
That's correct @dnadales, the set equality tell us that exactly the keys that need to sign have signed (no more, no less). Then, provided that the signatures validate, the witnesses are good.

We can't replace that set comprehension with `\dom{(\txwits{tx})}`, though, since the witnesses provide pubkeys, and we are collecting the hashes of pubkeys.